community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/TransactionBoundQueryContext.scala

```diff
@@ -161,11 +161,17 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
   override def getPropertiesForRelationship(relId: Long): Iterator[Int] =
     JavaConversionSupport.asScala(transactionalContext.statement.readOperations().relationshipGetPropertyKeys(relId))
 
-  override def isLabelSetOnNode(label: Int, node: Long): Boolean =
-    transactionalContext.statement.readOperations().nodeHasLabel(node, label)
+  override def isLabelSetOnNode(label: Int, node: Long): Boolean = {
+    var nodeCursor: NodeCursor = null;
+    try {
+      nodeCursor = cursors.allocateNodeCursor()
+      reads().singleNode(node, nodeCursor)
+      if (!nodeCursor.next()) false
+      else nodeCursor.labels().contains(label)
+    }
```

Comment:
Forgot to close the cursor. If we go this route, we will not some utility for try-with-resources in scala.

Reply:
we already have that, however require lambda. Thought we'll do it without allocation here

Reply:
1) Cool, I didn't know we had that... should be moved out of the frontend module though.
2) Ok, I like less allocation, so let's to that.